src/legacy/core_plugins/elasticsearch/index.js

```diff
@@ -46,7 +45,10 @@ export default function (kibana) {
 
       return Joi.object({
         enabled: Joi.boolean().default(true),
-        url: Joi.string().uri({ scheme: ['http', 'https'] }).default('http://localhost:9200'),
+        sniffOnStart: Joi.boolean().default(false),
+        sniffInterval: Joi.number().allow(false).default(false),
+        sniffOnConnectionFault: Joi.boolean().default(false),
+        hosts: Joi.array().items(Joi.string().uri({ scheme: ['http', 'https'] })).single().default('http://localhost:9200'),
```

Comment:
I'm sure this has been hashed out, but why go through all of the effort of a breaking change when we could just change `url` to have this same `Joi` definition and make it a noop for all users? Users that want to take advantage of it are then free to do so.

Reply:
It's also consistent with the HTTP exporter for Monitoring in Elasticsearch.

> and to support a s/url/hosts in the config (the Joi.array().items().single() part)

Sounds good. As long as this happens, then I'm all for consistency.